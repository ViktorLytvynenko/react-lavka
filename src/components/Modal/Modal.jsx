import style from "./Modal.module.scss";
import Buttons from "./Buttons/Buttons";
import closeButton from "../../img/closeButton.png";
import PropTypes from 'prop-types';


const Modal = (props) => {

    let themeStyle = {};
    if (props.theme === "danger") {
        themeStyle.header = style.modal_header_danger;
        themeStyle.body = style.modal_body_danger;
        themeStyle.buttons = style.modal_body_container_button_danger;
    }
    if (props.theme === "primary") {
        themeStyle.header = style.modal_header_primary;
        themeStyle.body = style.modal_body_primary;
        themeStyle.buttons = style.modal_body_container_button_primary;
    }
    return (
        <>
        <div className={style.modalContainer} onClick={props.handleClick}>
        <div className={style.modal} onClick={(event) => {event.stopPropagation()}}>
            <div className={`${style.modal_header} ${themeStyle.header}`}>
                <p>{props.text1}</p>
                {props.isClosed ?
                    <button onClick={props.handleClick}>
                        <img src={closeButton} alt="closedBtn"/>
                    </button> : null}
            </div>
            <div className={`${style.modal_body} ${themeStyle.body}`}>
                <p>{props.text2}</p>
                <p>Підтвердіть Ваш вибір</p>
                <Buttons handleClick={props.handleClick} confirm={props.confirm} themeStyle={themeStyle.buttons}/>
            </div>
        </div>
        </div>
        </>
    )
}

Modal.propTypes = {
    theme: PropTypes.oneOf(['danger', 'primary']).isRequired,
    text1: PropTypes.string.isRequired,
    isClosed: PropTypes.bool,
    text2: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
    confirm: PropTypes.func.isRequired,
};

Modal.defaultProps = {
    isClosed: false,
};

export default Modal