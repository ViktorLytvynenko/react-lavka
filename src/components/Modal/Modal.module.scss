.modalContainer {
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #0000006b;
  position: fixed;
}
.modal {
  width: 500px;
  height: 300px;
  background-color: #F0FFFF;
  color: white;
  border-radius: 10px;
  overflow: hidden;
  &_header {
    width: 460px;
    height: 20%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 20px;
    &_danger{
      background-color: #D64331;
    }
    &_primary{
      background-color: #65A6D1;
    }
    p {
      font-size: 20px;
      font-weight: bold;
    }
    button {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 35px;
      height: 35px;
      cursor: pointer;
      background: none;
      border: none;
      border-radius: 5px;
      img {
        width: 100%;
      }
    }
  }
  &_body {
    width: 100%;
    height: 80%;
    &_danger{
      background-color: #EA4935;
    }
    &_primary{
      background-color: darkolivegreen;
    }
    p {
      text-align: center;
      padding: 40px 30px 0 30px;
    }
    &_container {
      width: 100%;
      display: flex;
      justify-content: center;
      column-gap: 10px;
      margin-top: 50px;
      &_button {
        width: 100px;
        height: 35px;
        cursor: pointer;
        background: #B53728;
        border: none;
        font-weight: 500;
        color: white;
        border-radius: 5px;
        &_danger{
          background-color: #B53728;
        }
        &_primary{
          background-color: dodgerblue;
        }
      }
    }
  }
}