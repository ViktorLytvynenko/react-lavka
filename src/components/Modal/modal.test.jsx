import Modal from "./Modal";
import {render, screen} from "@testing-library/react";

describe('Modal component', () => {
    it("Render Modal with text", () => {
        render(
            <Modal
                theme="danger"
                text1="1"
                text2="2"
                handleClick={()=>{}}
                confirm={()=>{}}
            />
        )
        expect(screen.getByText("1")).toBeInTheDocument()
        expect(screen.getByText("2")).toBeInTheDocument()
    })
    it("Modal snapshot", () => {
        let modal;
        modal = render(
            <Modal
                theme="danger"
                text1="1"
                text2="2"
                handleClick={()=>{}}
                confirm={()=>{}}
            />
        )
        expect(modal).toMatchSnapshot()
    })
})