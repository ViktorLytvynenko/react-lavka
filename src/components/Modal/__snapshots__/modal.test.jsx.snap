// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Modal snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modalContainer"
      >
        <div
          class="modal"
        >
          <div
            class="modal_header modal_header_danger"
          >
            <p>
              1
            </p>
          </div>
          <div
            class="modal_body modal_body_danger"
          >
            <p>
              2
            </p>
            <p>
              Підтвердіть Ваш вибір
            </p>
            <div
              class="modal_body_container"
            >
              <button
                class="modal_body_container_button modal_body_container_button_danger"
              >
                Підтверджую
              </button>
              <button
                class="modal_body_container_button modal_body_container_button_danger"
              >
                Відмінити
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modalContainer"
    >
      <div
        class="modal"
      >
        <div
          class="modal_header modal_header_danger"
        >
          <p>
            1
          </p>
        </div>
        <div
          class="modal_body modal_body_danger"
        >
          <p>
            2
          </p>
          <p>
            Підтвердіть Ваш вибір
          </p>
          <div
            class="modal_body_container"
          >
            <button
              class="modal_body_container_button modal_body_container_button_danger"
            >
              Підтверджую
            </button>
            <button
              class="modal_body_container_button modal_body_container_button_danger"
            >
              Відмінити
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
